hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/tools/federation/RouterAdmin.java

```diff
@@ -699,32 +697,46 @@ public boolean addMount(AddMountAttributes addMountAttributes)
     // Get the existing entry
     MountTableManager mountTable = client.getMountTableManager();
     MountTable existingEntry = getMountEntry(mount, mountTable);
-    MountTable existingOrNewEntry =
-        addMountAttributes.getNewOrUpdatedMountTableEntryWithAttributes(existingEntry);
-    if (existingOrNewEntry == null) {
+    if (existingEntry != null) {
+      System.err.println("MountTable " + mount + " already exists.");
       return false;
     }
 
-    if (existingEntry == null) {
-      AddMountTableEntryRequest request = AddMountTableEntryRequest
-          .newInstance(existingOrNewEntry);
-      AddMountTableEntryResponse addResponse = mountTable.addMountTableEntry(request);
-      boolean added = addResponse.getStatus();
-      if (!added) {
-        System.err.println("Cannot add mount point " + mount);
-      }
-      return added;
-    } else {
-      UpdateMountTableEntryRequest updateRequest =
-          UpdateMountTableEntryRequest.newInstance(existingOrNewEntry);
-      UpdateMountTableEntryResponse updateResponse =
-          mountTable.updateMountTableEntry(updateRequest);
-      boolean updated = updateResponse.getStatus();
-      if (!updated) {
-        System.err.println("Cannot update mount point " + mount);
-      }
-      return updated;
+    MountTable mountEntry = addMountAttributes.getMountTableEntryWithAttributes();
+    AddMountTableEntryRequest request = AddMountTableEntryRequest.newInstance(mountEntry);
+    AddMountTableEntryResponse addResponse = mountTable.addMountTableEntry(request);
+    boolean added = addResponse.getStatus();
+    if (!added) {
+      System.err.println("Cannot add mount point " + mount);
     }
+    return added;
+  }
+
+  /**
+   * Return the map from namespace to destination.
+   * @param nss input namespaces.
+   * @param destinations input destinations.
+   * @return one map from namespace to destination.
+   */
+  public static Map<String, String> getDestMap(String[] nss, String[] destinations) {
+    assert destinations.length == 1 || destinations.length == nss.length;
```

Comment:
I would say rather than the assert use the if else and throw a proper message, ``isInvalidDestinations(String[] nss, String[] destinations)`` can be refactored for it, or pass additional ``boolean`` there to specify if we need to do a print or not